display-device-location-with-nmea-data-sources/src/main/java/com/esri/arcgismaps/sample/displaydevicelocationwithnmeadatasources/MainActivity.kt

```diff
@@ -246,38 +242,21 @@ class MainActivity : AppCompatActivity() {
      */
     private fun collectSatelliteChanges() = lifecycleScope.launch {
         nmeaLocationDataSource.satellitesChanged.collect { nmeaSatelliteInfoList ->
-            val uniqueSatelliteIDs = mutableListOf<Int>()
             var satelliteSystems = ""
             // set the text of the satellite count label
             satelliteCountTV.text = getString(R.string.satellite_count) + nmeaSatelliteInfoList.size
             // get the system of the first satellite
-            when (nmeaSatelliteInfoList.first().system) {
-                NmeaGnssSystem.Bds -> {
-                    satelliteSystems = "BDS"
-                }
-                NmeaGnssSystem.Galileo -> {
-                    satelliteSystems = "Galileo"
-                }
-                NmeaGnssSystem.Glonass -> {
-                    satelliteSystems = "Glonass"
-                }
-                NmeaGnssSystem.Gps -> {
-                    satelliteSystems = "GPS"
-                }
-                NmeaGnssSystem.NavIc -> {
-                    satelliteSystems = "NavIc"
-                }
-                NmeaGnssSystem.Qzss -> {
-                    satelliteSystems = "Qzss"
-                }
-                NmeaGnssSystem.Unknown -> {
-                    satelliteSystems = "Unknown"
-                }
+            satelliteSystems = when (nmeaSatelliteInfoList.first().system) {
```

Comment:
Instead of having `var satelliteSystems` above, the val keyword could be added here
```suggestion
            val satelliteSystems = when (nmeaSatelliteInfoList.first().system) {
```